src/base/bittorrent/session.cpp

```diff
@@ -358,97 +360,18 @@ Session::Session(QObject *parent)
-    pack.set_str(lt::settings_pack::peer_fingerprint, peerId);
-    pack.set_bool(lt::settings_pack::listen_system_port_fallback, false);
-    pack.set_str(lt::settings_pack::user_agent, USER_AGENT);
-    pack.set_bool(lt::settings_pack::use_dht_as_fallback, false);
-    // Speed up exit
-    pack.set_int(lt::settings_pack::stop_tracker_timeout, 1);
-    pack.set_int(lt::settings_pack::auto_scrape_interval, 1200); // 20 minutes
-    pack.set_int(lt::settings_pack::auto_scrape_min_interval, 900); // 15 minutes
-    pack.set_int(lt::settings_pack::connection_speed, 20); // default is 10
-    pack.set_bool(lt::settings_pack::no_connect_privileged_ports, false);
-    // libtorrent 1.1 enables UPnP & NAT-PMP by default
-    // turn them off before `lt::session` ctor to avoid split second effects
-    pack.set_bool(lt::settings_pack::enable_upnp, false);
-    pack.set_bool(lt::settings_pack::enable_natpmp, false);
-    pack.set_bool(lt::settings_pack::upnp_ignore_nonrouters, true);
-
-#if (LIBTORRENT_VERSION_NUM < 10200)
-    // Disable support for SSL torrents for now
-    pack.set_int(lt::settings_pack::ssl_listen, 0);
-    // To prevent ISPs from blocking seeding
-    pack.set_bool(lt::settings_pack::lazy_bitfields, true);
-    // Disk cache pool is rarely tested in libtorrent and doesn't free buffers
-    // Soon to be deprecated there
-    // More info: https://github.com/arvidn/libtorrent/issues/2251
-    pack.set_bool(lt::settings_pack::use_disk_cache_pool, false);
-#endif
-
-    configure(pack);
-
-    m_nativeSession = new lt::session {pack, LTSessionFlags {0}};
-    m_nativeSession->set_alert_notify([this]()
-    {
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))
-        QMetaObject::invokeMethod(this, &Session::readAlerts, Qt::QueuedConnection);
-#else
-        QMetaObject::invokeMethod(this, "readAlerts", Qt::QueuedConnection);
-#endif
-    });
-
-    configurePeerClasses();
-
-    // Enabling plugins
-    //m_nativeSession->add_extension(&lt::create_metadata_plugin);
-    m_nativeSession->add_extension(&lt::create_ut_metadata_plugin);
-    if (isPeXEnabled())
-        m_nativeSession->add_extension(&lt::create_ut_pex_plugin);
-    m_nativeSession->add_extension(&lt::create_smart_ban_plugin);
-
-    LogMsg(tr("Peer ID: ") + QString::fromStdString(peerId));
-    LogMsg(tr("HTTP User-Agent is '%1'").arg(USER_AGENT));
-    LogMsg(tr("DHT support [%1]").arg(isDHTEnabled() ? tr("ON") : tr("OFF")), Log::INFO);
-    LogMsg(tr("Local Peer Discovery support [%1]").arg(isLSDEnabled() ? tr("ON") : tr("OFF")), Log::INFO);
-    LogMsg(tr("PeX support [%1]").arg(isPeXEnabled() ? tr("ON") : tr("OFF")), Log::INFO);
-    LogMsg(tr("Anonymous mode [%1]").arg(isAnonymousModeEnabled() ? tr("ON") : tr("OFF")), Log::INFO);
-    LogMsg(tr("Encryption support [%1]").arg((encryption() == 0) ? tr("ON") :
-        ((encryption() == 1) ? tr("FORCED") : tr("OFF"))), Log::INFO);
+    initializeNativeSession();
+    configureComponents();
```

Comment:
Are you sure there is same logic in both cases it is used?

Reply:
Yes, I checked the details and this should work.